### ISSUE TITLE
Disable the "x is not accessed" hint when the function in question is marked as an abstractmethod

### ISSUE THREAD

**Issue by dob9601** [microsoft/pyright#1742]
**Is your feature request related to a problem? Please describe.**
When creating an abstract method (using the `@abstractmethod` decorator in `abc`), function arguments are never used due to the empty function body, for example:
```py
@abc.abstractmethod
def foo(self, bar):
    ...
```
would lead to pyright notifying you that `self` and `bar` aren't used

**Describe the solution you'd like**
Suppress this diagnostic in the case of it applying to an abstractmethod

**Comment by erictraut** [comment]
This is a duplicate of https://github.com/microsoft/pyright/issues/1541.

The current behavior is "as designed", and we don't have any current plans to change it.

I'm curious what editor you're using? The problem that some users are facing is that their editor is interpreting this hint incorrectly and displaying the hint in a manner where it appears as a warning or an error. This hint is intended to cause the editor to display the text using a subtle visual cue like "grayed out" text. Some editors are reportedly doing things like underlining the text and displaying prominently displaying warning text. I consider this a bug in the editor — or at least a significant design flaw. VS Code handles this case correctly. If you are using an editor other than VS Code, and the editor is not handling this hint correctly, please report the problem to the maintainer of your editor so they fix it.

**Comment by dob9601** [comment]
That's fair - I'm using pyright via coc-pyright with vim which correctly displays it in light blue (representing info, as opposed to a warning/error)
